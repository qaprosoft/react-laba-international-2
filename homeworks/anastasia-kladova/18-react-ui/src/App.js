import './index.css';
import AvatarsBox from './components/AvatarsBox/AvatarsBox';
import Button from './components/Buttons/Button/Button';
import {useState} from 'react';
import {getNewAvatar} from './utils/getNewAvatar';
import Loader from './components/Loader/Loader';
import LoaderContainer from './components/LoaderContainer.jsx/LoaderContainer';

const App = () => {
  const [isLoading, setIsLoading] = useState(false);
  const [avatars, setAvatars] = useState([]);

  const avatarsLength = avatars.length;

  const refreshAll = async () => {
    setIsLoading(true);

    if (!avatarsLength) return;
    const data = await getNewAvatar(avatarsLength);

    setAvatars([data]);
    setIsLoading(false);
  };
  console.log(isLoading);
  return (
    <main className="main">
      <section className="avatars">
        {isLoading ? (
          <LoaderContainer />
        ) : (
          <div className="container avatars__container">
            <AvatarsBox
              avatars={avatars}
              setAvatars={setAvatars}
              isLoading={isLoading}
              setIsLoading={setIsLoading}
            />
            <Button
              handler={refreshAll}
              isLoading={isLoading}
              setIsLoading={setIsLoading}
            >
              Refresh All
            </Button>
          </div>
        )}
      </section>
    </main>
  );
};

export default App;
