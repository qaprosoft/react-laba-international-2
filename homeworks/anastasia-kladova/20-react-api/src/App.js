import {useContext} from 'react';
import Form from './components/Form/Form';
import {Context} from './contexts/AppContext/AppContext';
import TodoList from './components/TodoList/TodoList';
import Button from './components/Buttons/Button/Button';
import {saveDataToStorage} from './utils/saveDataToStorage';
import Modal from './components/Modal/Modal';
import {phrases} from './constants/constants';

const App = () => {
  const {
    isShowDeleteModal,
    setIsShowDeleteModal,
    todos,
    setTodos,
    todoToDelete,
    setTodoToDelete,
  } = useContext(Context);

  const deleteCompletedTodo = () => {
    const newTodos = todos.filter(todo => todo.isCompleted !== true);
    setTodos(newTodos);
    saveDataToStorage(newTodos);
  };

  const deleteTodo = id => {
    const newTodos = todos.filter(todo => todo.id !== id);
    setTodos(newTodos);
    saveDataToStorage(newTodos);
    setTodoToDelete(null);
    setIsShowDeleteModal(false);
  };

  console.log(isShowDeleteModal);

  return (
    <main className="main">
      <section className="todo">
        <div className="container todo__container">
          <Form />
          <TodoList />
          <Button
            type="button"
            btnText="Delete completed"
            onClickHandler={() => deleteCompletedTodo()}
          />
        </div>
      </section>
      {isShowDeleteModal && (
        <Modal
          modalText={phrases.DELETEMODAL}
          deleteHandler={() => deleteTodo(todoToDelete)}
        />
      )}
    </main>
  );
};

export default App;
