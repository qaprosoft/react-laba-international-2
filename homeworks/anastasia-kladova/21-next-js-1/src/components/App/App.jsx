

import {useState} from 'react';

import AvatarsBox from '../AvatarsBox/AvatarsBox';
import Button from '../Buttons/Button/Button';
import { getNewAvatar } from '@/utils/getNewAvatar';

const App = ({avatarsFromServer}) => {
  const [avatars, setAvatars] = useState(avatarsFromServer);
  const [isLoading, setIsLoading] = useState(false);
  const avatarsLength = avatars.length;


  const refreshAll = async () => {
    setIsLoading(true);

    if (!avatarsLength) return;
    const data = await getNewAvatar(avatarsLength);
    setAvatars(data);

    setIsLoading(false);
  };

  return (
    <main className="main">
      <section className="avatars">
        <div className="container avatars__container">
          <AvatarsBox
            avatars={avatars}
            setAvatars={setAvatars}
            isLoading={isLoading}
            setIsLoading={setIsLoading}
          />
          <Button handler={refreshAll} isLoading={isLoading}>
            Refresh All
          </Button>
        </div>
      </section>
    </main>
  );
};

export default App;
