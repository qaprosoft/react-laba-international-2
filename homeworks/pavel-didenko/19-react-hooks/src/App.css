.main-container {
  background-image: url('./assets/img/backgrounds/background.png');
  min-height: 100vh;
  background-size: cover;
}

.tasks-section {
  display: flex;
  flex-direction: column;
  gap: 0rem;
  justify-content: center;
  align-items: center;
  margin-top: 2.1rem;
}