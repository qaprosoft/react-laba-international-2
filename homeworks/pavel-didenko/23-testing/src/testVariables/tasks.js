export const tasks1 = [
  {
    id: 1,
    taskText: 'someTask1',
    completed: false,
  },
  {
    id: 2,
    taskText: 'someTask2',
    completed: false,
  },
  {
    id: 3,
    taskText: 'someTask3',
    completed: false,
  },
  {
    id: 4,
    taskText: 'someTask4',
    completed: true,
  },
];

export const tasks2 = [
  {
    id: 1,
    taskText: 'someTask1',
    completed: false,
  },
  {
    id: 2,
    taskText: 'someTask2',
    completed: false,
  },
  {
    id: 3,
    taskText: 'someTask3',
    completed: false,
  },
];