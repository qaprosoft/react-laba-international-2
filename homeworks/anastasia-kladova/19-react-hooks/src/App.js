import './App.css';
import Form from './components/Form/Form';
import TodoItem from './components/TodoItem/TodoItem';

const App = () => {
  return (
    <main className="main">
      <section className="todo">
        <div className="container todo__container">
          <Form />
          <TodoItem />
        </div>
      </section>
    </main>
  );
};

export default App;
