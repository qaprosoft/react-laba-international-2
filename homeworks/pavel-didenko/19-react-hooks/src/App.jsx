import './App.css';
import Task from './components/Task/Task';
import TaskCreator from './components/TaskCreator/TaskCreator';
import {useState, useEffect} from 'react';
import ClearCompletedTasksButton from './components/ClearCompletedTasksButton/ClearCompletedTasksButton';

function App() {
  const [tasks, setTasks] = useState([]);

  useEffect(() => {
    const storedTasks = localStorage.getItem('tasks');
    if (storedTasks) {
      setTasks(JSON.parse(storedTasks));
    }
  }, []);

  useEffect(() => {
    localStorage.setItem('tasks', JSON.stringify(tasks));
  }, [tasks]);

  function createTask(newTaskText) {
    setTasks(tasks => [
      ...tasks,
      {id: Date.now(), taskText: newTaskText, completed: false},
    ]);
  }

  function modifyTasks(id, text) {
    setTasks(tasks => {
      return tasks.map((task) => {
        if (task.id !== id) {
          return task;
        } else {
          task.taskText = text;
          return task;
        }
      });
    });
  }

  function setCompletedTask(id) {
    setTasks(tasks => {
      return tasks.map((task) => {
        if (task.id !== id) {
          return task;
        } else {
          task.completed = !task.completed;
          return task;
        }
      });
    });
  }

  function removeTask(id) {
    setTasks(tasks.filter(task => task.id !== id));
  }

  function removeCompletedTasks() {
    setTasks(tasks.filter((task) => !task.completed))
  }

  return (
    <div className="main-container">
      <TaskCreator createTask={createTask} tasks={tasks} />
      <div className="tasks-section">
        {tasks.map((task, index) => {
          return (
            <Task
              taskText={task.taskText}
              key={task.id}
              id={task.id}
              completed={task.completed}
              modifyTasks={modifyTasks}
              removeTask={removeTask}
              setCompletedTask={setCompletedTask}
              setTasks={setTasks}
            />
          );
        })}
        <ClearCompletedTasksButton
          removeCompletedTasks={removeCompletedTasks}
          tasks={tasks}
        />
      </div>
    </div>
  );
}

export default App;
