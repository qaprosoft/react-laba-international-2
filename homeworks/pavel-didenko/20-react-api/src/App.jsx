import './App.css';
import Task from './components/Task/Task';
import TaskCreator from './components/TaskCreator/TaskCreator';
import { useReducer, useCallback} from 'react';
import reducer from './functions/reducer';
import { MainContext } from './contexts/mainContext';
import ClearCompletedTasksButton from './components/ClearCompletedTasksButton/ClearCompletedTasksButton';
import useSaveToLocalStorage from './hooks/useSaveToLocalStorage';

const App = function () {
  const [state, dispatch] = useReducer(reducer, []);

  const tasksUploader = useCallback(() => {
    return state.map((task) => {
      return (
        <Task
          taskText={task.taskText}
          key={task.id}
          id={task.id}
          completed={task.completed}
          modifyTasks={modifyTasks}
          removeTask={removeTask}
          setCompletedTask={setCompletedTask}
          state={state}
        />
      );
    });
  }, [state]);

  useSaveToLocalStorage(state, dispatch);

  function createTask(newTaskText) {
    dispatch({type: 'task_added', text: newTaskText});
  }

  function modifyTasks(id, text) {
    dispatch({type: 'task_modified', 'id': id, text: text});
  }

  function setCompletedTask(id) {
    dispatch({type: 'task_completed', 'id': id});
  }

  function removeTask(id) {
    dispatch({type: 'task_removed', 'id': id});
  }

  function removeCompletedTasks() {
    dispatch({type: 'remove_completed_tasks'});
  }

  return (
    <MainContext.Provider
      value={{
        createTask,
        modifyTasks,
        setCompletedTask,
        removeTask,
        removeCompletedTasks,
      }}
    >
      <div className="main-container">
        <TaskCreator state={state} />
        <div className="tasks-section">
          {tasksUploader()}
          <ClearCompletedTasksButton
            removeCompletedTasks={removeCompletedTasks}
            tasks={state}
          />
        </div>
      </div>
    </MainContext.Provider>
  );
};

export default App;
