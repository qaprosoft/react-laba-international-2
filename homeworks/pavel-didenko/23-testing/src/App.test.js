import renderer from 'react-test-renderer';
import App from './App';
import {render, screen, fireEvent} from '@testing-library/react';
import {tasks1} from './testVariables/tasks';

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Has task length warning message', () => {
  render(<App />);
  screen.getByText('Task length must be from 0 to 33 characters');
});

test('Renders tasks from localStorage', () => {
  localStorage.setItem('tasks', JSON.stringify(tasks1));
  render(<App />);
  screen.getByDisplayValue('someTask1');
  screen.getByDisplayValue('someTask2');
  screen.getByDisplayValue('someTask3');
  screen.getByDisplayValue('someTask4');
  localStorage.clear();
});