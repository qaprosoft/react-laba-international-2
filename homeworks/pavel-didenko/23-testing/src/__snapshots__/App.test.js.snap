// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="task-creator-wrapper"
  >
    <div>
      <input
        onChange={[Function]}
        placeholder="Create-Todo-Task"
      />
      <p
        className="task-creator__warning"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        Task length must be from 0 to 33 characters
      </p>
    </div>
    <button
      className="add-task-button"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="tasks-section"
  >
    <button
      className="clear-completed-tasks"
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      Clear completed tasks
    </button>
  </div>
</div>
`;
